assembly_and_binning/tutorial_assembly_and_binning.md

```diff
@@ -0,0 +1,634 @@
+
+Removing output files of failed job run_checkm_lineage_wf since they might be corrupted:
+SRR8859675/binning/DASTool/checkm/storage/tree/concatenated.fasta
+Shutting down, this might take some time.
+Exiting because a job execution failed. Look above for error message
+Note the path to the log file for debugging.
+Documentation is available at: https://metagenome-atlas.readthedocs.io
+Issues can be raised at: https://github.com/metagenome-atlas/atlas/issues
+The code used to generate one or several output files has changed:
+    To inspect which output files have changes, run 'snakemake --list-code-changes'.
+    To trigger a re-run, use 'snakemake -R $(snakemake --list-code-changes)'.
+The input used to generate one or several output files has changed:
+    To inspect which output files have changes, run 'snakemake --list-input-changes'.
+    To trigger a re-run, use 'snakemake -R $(snakemake --list-input-changes)'.
+Complete log: .snakemake/log/2022-07-22T085856.769525.snakemake.log
+[Atlas] CRITICAL: Command 'snakemake --snakefile /home/stamps2022/.conda/envs/atlas/lib/python3.8/site-packages/atlas/workflow/Snakefile --directory /home/stamps2022/atlas_workflow --jobs 8 --rerun-incomplete --configfile '/home/stamps2022/atlas_workflow/config.yaml' --nolock   --use-conda --conda-prefix /home/stamps2022/atlas_workflow/databases/conda_envs    --resources mem=27 mem_mb=28585 java_mem=23   --scheduler greedy  binning   ' returned non-zero exit status 1.
+```
+
+We see from the top of this message that snakemake is encountering an error.
+Let's trouble shoot this error and figure out what happened.
+
+In the error message, we see `&> SRR8859675/logs/binning/DASTool/checkm.log`. 
+This is where the log message for this command is written.
+Let's look at this error message and try and figure out what happened.
+
+
+**Challenge 4**
+Look at the stderror log message that was saved.
+What do you think happened?
+What process did you go through to figure out what happened?
+
+```
+less SRR8859675/logs/binning/DASTool/checkm.log
+```
+
+<details>
+  <summary>Challenge Answer</summary>
+It looks as though `pplacer` failed. 
+We can copy the relevant parts of the error message and google them to see if anyone has encountered these problems before. 
+If you google:
+
+```
+Uncaught exception: Sys_error("SRS049959/binning/DASTool/checkm/storage/tree/concatenated.pplacer.json: No such file or directory")
+Fatal error: exception Sys_error("SRS049959/binning/DASTool/checkm/storage/tree/concatenated.pplacer.json: No such file or directory")
+```
+
+You might come upon [this](https://github.com/Ecogenomics/CheckM/issues/37) github issue. 
+
+The key turns out to be that `Killed` message in the output above, which generally tells you that the computer ran out of memory at this step. This is because we "only" have 16 GB of RAM on these computers, and pplacer requires more.
```

Comment:
```suggestion
The key turns out to be that `Killed` message in the output above, which generally tells you that the computer ran out of memory at this step. This is because we "only" have 29 GB of RAM on the Jetstream computers (and less on binder!), and pplacer requires more.
```